tests/conftest.py

```diff
@@ -281,3 +284,84 @@ def pytest_runtest_protocol(item):
             ihook.pytest_runtest_logfinish(nodeid=nodeid, location=item.location)
 
         return True
+
+
+# source code fixtures
+
+
+def _run(cmd):
+    return subprocess.check_output(cmd, shell=True)
+
+
+@contextlib.contextmanager
+def create_package(directory, pyproject, setup):
+    package_dir = os.path.join(directory, "mypackage")
+    os.mkdir(package_dir)
+
+    pyproject_file = os.path.join(package_dir, "pyproject.toml")
+    with open(pyproject_file, "wb") as f:
+        f.write(pyproject.encode("utf-8"))
+
+    setup_file = os.path.join(package_dir, "setup.py")
+    with open(setup_file, "wb") as f:
+        f.write(setup.encode("utf-8"))
+
+    _ = os.path.join(package_dir, "mypackage")
+    os.mkdir(_)
+    with open(os.path.join(_, "__init__.py"), "wb") as f:
+        f.write('"0.0.1"'.encode("utf-8"))
+
+    cwd = os.getcwd()
+    os.chdir(package_dir)
+
+    try:
+        _run("git init")
+        _run("git config --local user.name user")
+        _run("git config --local user.email user@company.com")
+        _run("git add .")
+        _run("git commit -m init")
+        _run("git remote add origin https://github.com/companydotcom/repo.git")
+
+        yield package_dir
+    finally:
+        os.chdir(cwd)
+
+        try:
+            shutil.rmtree(package_dir)
+        except OSError:
+            pass
+
+
+@pytest.fixture
+def temp_dir():
```

Comment:
pytest already has a built-in `tmpdir` fixture.

https://docs.pytest.org/en/6.2.x/tmpdir.html

Reply:
Super! Thanks!